mirascope/core/base/prompt.py

```diff
@@ -323,13 +337,13 @@ def __call__(
         | Callable[_P, BaseDynamicConfig]
         | Callable[_P, Awaitable[BaseDynamicConfig]],
     ) -> (
-        Callable[_P, list[BaseMessageParam]]
-        | Callable[_P, Awaitable[list[BaseMessageParam]]]
+        Callable[_P, str | list[BaseMessageParam]]
```

Comment:
i'm confused by this typing, wouldn't the decorator return only `list[BaseMessageParam]` or `Awaitable[list[BaseMessageParam]]`? In what case does it return `str`?